Feat: Debug mode and modify logLevel

LeonFLK opened the pull request:
## fixes KILTProtocol/ticket#496


Changes the default logLevel from Debug to Error, with the option to enable Debug mode with env variable DEBUG.

Adds new exported function to change the logLevel of all factories to the specified level.

## How to test:

start with or without DEBUG env set to true, call modifyLogLevel, see many logs or not.

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [x] I have documented the changes (where applicable)

LeonFLK commented:
tests were failing due to unreached 80% function coverage

wischli commented on src/config/ConfigLog.ts:
```diff
@@ -11,19 +11,44 @@ import {
+/**
+ *  Changes all existing Loggers of our default Factory with id 0 to the intended Level.
+ *
+ * @param level The intended LogLevel.
```
Could potentially add that this ranges from 0 to 5 but do as you wish.

LeonFLK replied:
Yes good addition.

I wish though the LogLevel enum had this kind of information at runtime.

wischli commented:
> tests were failing due to unreached 80% function coverage

Yeah I removed the config from the coverage.

LeonFLK commented:
Tests run ~8% faster with LogLevel reduced from Debug to Error yey

LeonFLK commented:
> > tests were failing due to unreached 80% function coverage
> 
> Yeah I removed the config from the coverage.

I added Config.spec.ts that checks the modify function and the default level.
